gapis/api/vulkan/api/coherent_memory.api

```diff
@@ -57,19 +57,14 @@ sub void readCoherentMemory(ref!DeviceMemoryObject memory, VkDeviceSize readOffs
 }
 
 sub void readMemoryInBuffer(ref!BufferObject buffer, VkDeviceSize readOffset, VkDeviceSize readSize) {
-  mem := buffer.Memory
-  if mem != null {
-    readBegin := buffer.MemoryOffset + readOffset
-    actualReadSize := switch as!u64(readSize) {
-      case 0xFFFFFFFFFFFFFFFF:
-        buffer.Info.Size - readOffset
-      default:
-        readSize
-    }
-    readEnd := readBegin + actualReadSize
-    readCoherentMemory(mem, readBegin, actualReadSize)
-    if len(mem.Data) != 0 {
-      read(mem.Data[readBegin : readEnd])
+  memPieces := getBufferBoundMemoryPiecesInRange(buffer, readOffset, readSize)
+  for _, _, mp in memPieces {
+    readCoherentMemory(mp.DeviceMemory, mp.MemoryOffset, mp.Size)
+    // On the trace side, 'Data' pool won't actually be created, so its length
```

Comment:
Instead of this way, since data is marked as "spy_disable" can we just not generate the read intrinsic for this case?

We already handle other cases like this, so I think cutting out the read intrinsic for spy_disabled is probably a cleaner approach than a check like this.

Reply:
Fixed in 2a680ed